public/app/app.ts

```diff
@@ -213,22 +214,30 @@ export class GrafanaApp {
       extensionsRegistry.register({
         pluginId: 'grafana',
         extensionConfigs: getCoreExtensionConfigurations(),
+        exportedComponentConfigs: [],
       });
 
+      const exportedComponentsRegistry = new ExportedComponentsRegistry();
+
       if (contextSrv.user.orgRole !== '') {
         // The "cloud-home-app" is registering banners once it's loaded, and this can cause a rerender in the AppChrome if it's loaded after the Grafana app init.
         // TODO: remove the following exception once the issue mentioned above is fixed.
         const awaitedAppPluginIds = ['cloud-home-app'];
         const awaitedAppPlugins = Object.values(config.apps).filter((app) => awaitedAppPluginIds.includes(app.id));
         const appPlugins = Object.values(config.apps).filter((app) => !awaitedAppPluginIds.includes(app.id));
 
-        preloadPlugins(appPlugins, extensionsRegistry);
-        await preloadPlugins(awaitedAppPlugins, extensionsRegistry, 'frontend_awaited_plugins_preload');
+        preloadPlugins(appPlugins, extensionsRegistry, exportedComponentsRegistry);
+        await preloadPlugins(
+          awaitedAppPlugins,
+          extensionsRegistry,
+          exportedComponentsRegistry,
```

Comment:
Once all extension types have been moved to their own registries, we can pass `registries: Array<Registry>` rather than using this approach.